### PR TITLE
[TIR] Fix compute inline for dynamic loop and nontrivial store indices

### PR DESCRIPTION
**Pull request by wrongtest-intellif**
Hi, the change wants to fix https://github.com/apache/tvm/issues/14364 where our `compute-inline` validity check failed. 
- Do not depend on undefined var counting, since dynamic loop extents may also be counted. The fast path just check the store indices should be the same with the block iter vars, which would cover the most cases.
- Add fallback implementation for store indices which are bijective wrt block iter vars, which is already utilized in `reverse-compute-inline` implementation.
- The change do not check the inlined producer's region cover the consumer. Refer to unittest `test_compute_inline_out_of_bound_consumer`, the previous behavior is not changed in this MR.

### CODE REVIEW

**Comment by yzh119** [comment]
Hi @wrongtest-intellif , this PR looks good to me, we can still use `GetNumUndefinedNonpointerVars` to check if we can avoid visiting buffer shape/strides in the `UndefinedVars` function (I have created another PR #https://github.com/apache/tvm/pull/14502 for that), what do you think?

**Comment by wrongtest-intellif** [comment]
> Hi @wrongtest-intellif , this PR looks good to me, we can still use `GetNumUndefinedNonpointerVars` to check if we can avoid visiting buffer shape/strides in the `UndefinedVars` function (I have created another PR ##14502 for that), what do you think?

Thank you for the remind! It seems great to rely on #14502 to solve the dynamic buffer issue and then I could go on for the 
patch when the write indices are not in the ordered form.

File: tests/python/unittest/test_tir_schedule_compute_inline.py [hunk 1]
```diff
@@ -894,6 +894,98 @@ def test_compute_inline_multi_consumer(use_block_name):
...
+            with T.block("C"):
+                vi, vj = T.axis.remap("SS", [i, j])
+                C[vi, vj] = A[vi, vj] * 2.0 + 1.0
+
+    sch = tir.Schedule(before, debug_mask="all")
+    sch.compute_inline(sch.get_block("B"))
+    tvm.ir.assert_structural_equal(after, sch.mod["main"])
+    verify_trace_roundtrip(sch=sch, mod=before)
+
+
```
Comment (yzh119) [inline tests/python/unittest/test_tir_schedule_compute_inline.py:988]: Adding another "nested block" case with variable loop extents in outer blocks is better.

**Review by yzh119 — COMMENTED** [review]
Hi @wrongtest-intellif , after I tried more dynamic shape cases I feel it unnecessary to keep the `GetNumUndefinedNonpointerVars` check, I'll close my PR and let's move on with your changes.

File: tests/python/unittest/test_tir_schedule_compute_inline.py (hunk 1 above)
Comment (yzh119) [reply tests/python/unittest/test_tir_schedule_compute_inline.py:988]: For example:

```python
@T.prim_func
def symbolic_shape_blockized(a: T.handle, b: T.handle):
    N = T.int32()
    A = T.match_buffer(a, (N, 32))
    B = T.match_buffer(b, (N, 32))
    for i in range(N):
        with T.block("outer"):
            vi = T.axis.spatial(N, i)
            temp = T.alloc_buffer((32), "float32", scope="local")
            for j in range(32):
                with T.block("T"):
                    vj = T.axis.spatial(32, j)
                    temp[vj] = A[vi, vj] + 1
            for j in range(32):
                with T.block("B"):
                    vj = T.axis.spatial(32, j)
                    B[vi, vj] = temp[vj] * A[vi, vj]
```

We should support computing inline the block `T`.

**Comment by yzh119** [comment]
Hi @wrongtest-intellif , any updates on this?

**Comment by wrongtest-intellif** [comment]
> Hi @wrongtest-intellif , any updates on this?

Sorry to be late, I would turn back to it soon.

**Pull request closed without merging**